### ISSUE TITLE
[Unity|Android] Crash on app start when Unity debug logging is enabled

### ISSUE THREAD

**Issue by Nyankoo** [firebase/quickstart-unity#391]
I'm getting the following crash when importing the firebase messaging plugin (.NET 4.x) and starting my app on a real device:

```
06-18 15:32:02.949  6063  6232 I firebase: Firebase App initializing app de.mycompany.myproject(default 1).
06-18 15:32:02.950  6063  6232 E CRASH   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0000001f
06-18 15:32:02.950  6063  6232 E CRASH   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-18 15:32:02.950  6063  6232 E CRASH   : Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a'
06-18 15:32:02.950  6063  6232 E CRASH   : Build fingerprint: 'samsung/zerofltexx/zeroflte:7.0/NRD90M/G920FXXU6ERF5:user/release-keys'
06-18 15:32:02.950  6063  6232 E CRASH   : Revision: '11'
```

Do I need to import any other packages (besides the libraries that the PlayServicesResolver pulls in)?

Any help would be appreciated!

**Comment by Nyankoo** [comment]
It seems the crash is related to setting the "Logging" option in Unity's "Other settings" to _Full_ instead of _ScriptOnly_.
Specifically, it's the "Log" option. All other options can be set to _Full_, but "Log" NEEDS to be set to _ScriptOnly_ or the app crashes.

This can be reproduced with the sample project.

The Unity version I'm using is 2018.4.2f1.

**Comment by patm1987** [comment]
Thanks for reporting this, I was able to repro.

I did find a workaround for the time being as well, if you set the scripting runtime to mono instead of il2cpp. We are tracking this now.

**Comment by stewartmiles** [comment]
We've figure out what's going on here.  We have C/C++ log redirection that sends logs from the C++ library through Unity's logger i.e we go C++-->C#-->UnityEngine.Debug.Log*().  When full stack trace logging is enabled Unity has some code that will attempt to unwind the stack which crashes when it hits out C++ module which definitely sounds like a bug in their unwinding logic since the system unwinders on Android / iOS can unroll this fine.  So to workaround this issue we're disabling C++->C# log redirection if full stack logging is enabled.  In practice on Android and iOS this makes no difference as all C++, Java & Obj-C library logs end up going to the console anyway.  In the editor, if full stack logging is enabled, C++ logs won't be visible unless running the editor from the command line to see stdout from the process.

The fix should land in our next release.

**Comment by alexames** [comment]
This should be fixed as of the latest release, 6.2.0
